File: examples/gateway-client.dart
import 'dart:io';

// OpenClaw Gateway Client (Dart)
Future<int> checkHealth(String baseUrl) async {
  final client = HttpClient()..connectionTimeout = const Duration(seconds: 5);
  try {
    final request = await client.getUrl(Uri.parse('$baseUrl/health'));
    final response = await request.close();
    return response.statusCode;
  } finally {
    client.close();
  }
}

Future<void> main(List<String> args) async {
  final url = args.isNotEmpty ? args[0] : 'http://localhost:8080';
  final status = await checkHealth(url);
  print('Gateway $url -> HTTP $status');
}

File: examples/gateway_client.nim
import std/[httpclient, os, strformat]

# OpenClaw Gateway Client (Nim)
proc checkHealth(baseUrl: string): int =
  var client = newHttpClient(timeout = 5000)
  try:
    let response = client.request(baseUrl & "/health", httpMethod = HttpGet)
    result = response.code.int
  except CatchableError:
    result = 0
  finally:
    client.close()

let url = if paramCount() > 0: paramStr(1) else: "http://localhost:8080"
echo &"Gateway {url} -> HTTP {checkHealth(url)}"
